ExampleMod/Common/Systems/ExampleWorldGenHookingSystem.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Mono.Cecil.Cil;
+using MonoMod.Cil;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Terraria;
+using Terraria.GameContent.Generation;
+using Terraria.IO;
+using Terraria.ModLoader;
+using Terraria.ModLoader.IO;
+using Terraria.WorldBuilding;
+
+namespace ExampleMod.Common.Systems
+{
+	// This ModSystem will demonstrate how to IL edit and Detour world generation passes
+	// Since world generation passes are anonymous methods (they don't have a name), they can't be edited the standard way (using IL_xx or On_xx)
+	public class ExampleWorldGenHookingSystem : ModSystem
+	{
+		// All of the registration should take place in load
+		// Generation pass hooks are unloaded manually, so no Unload method is needed
+		public override void Load() {
+			// IL editing the pyramids pass
+			GenPass pyramidsPass = WorldGen.VanillaGenPasses.First(t => t.Name == "Pyramids");// Locate the generation pass
```

Comment:
Need to call `FirstOrDefault` here if you want it to be error free

Reply:
Honestly, I'd prefer using `Single` and a cast, with no warning. Since no-one should be modifying the vanilla pass list anyway.